gammapy/datasets/io.py

```diff
@@ -230,6 +236,9 @@ def to_counts_hdulist(self, dataset, is_bkg=False):
         # adapt meta data
         table.meta.update(meta)
         hdulist["SPECTRUM"] = fits.BinTableHDU(table)
+
```

Comment:
Should we not update the creation date at the time of the writing (as made in `map.to_hdulist()`) rather using the date of the creation of the datasets?